Permissions/ViewController.m

```diff
@@ -629,25 +630,21 @@ - (void) tableView:(UITableView *) tableView didSelectRowAtIndexPath:(NSIndexPat
 
 #pragma mark - Address Book
 
-- (ABAddressBookRef) addressBook {
+- (CNContactStore *) addressBook {
   return _addressBook;
 }
 
-- (void) setAddressBook:(ABAddressBookRef) newAddressBook {
+- (void) setAddressBook:(CNContactStore *) newAddressBook {
   if (_addressBook != newAddressBook) {
-    if (_addressBook != NULL) {
-      CFRelease(_addressBook);
-    }
-    if (newAddressBook != NULL) {
-      CFRetain(newAddressBook);
-    }
     _addressBook = newAddressBook;
   }
 }
 
-void handleAddressBookChange(ABAddressBookRef addressBook,
-                             CFDictionaryRef info,
-                             void *context) {
+- (void) addressBookDidChange:(NSNotification *)notification {
+  [self handleAddressBookChange];
+}
+
+- (void) handleAddressBookChange; {
```

Comment:
Why did we need these changes?

Reply:
We need this to put the method to the callback when notification about contact store change did occur. This is an empty method because we don't really need to check contacts (We check only that alert form did appear and work with that).